pkg/generate/code/resource_reference_test.go

```diff
@@ -198,3 +198,265 @@ if ko.Spec.Routes != nil {
 return false || (ko.Spec.VPCRef != nil)`
 	assert.Equal(expected, code.ReferenceFieldsPresent(crd, "ko"))
 }
+
+func Test_ReferenceForField_SingleReference(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	g := testutil.NewModelForServiceWithOptions(t, "apigatewayv2",
+		&testutil.TestingModelOptions{
+			GeneratorConfigFile: "generator-with-reference.yaml",
+		})
+
+	crd := testutil.GetCRDByName(t, g, "Integration")
+	require.NotNil(crd)
+	expected :=
```

Comment:
Consider sticking this "expected" code into a separate file (perhaps under `testdata/`?) and compare its contents to the output below instead of embedding this value directly into the test

Reply:
With my latest changes the expected code is vastly smaller. Probably small enough we can consider it staying inline.